src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -555,19 +493,11 @@ impl ThinPool {
 
                 let usage = &status.usage;
 
-                // Kernel 4.19+ includes the kernel-set meta lowater value in
-                // thinpool status. For older kernels, use a default value.
-                let meta_lowater = status
-                    .meta_low_water
-                    .map(MetaBlocks)
-                    .unwrap_or(META_LOWATER_FALLBACK);
-                if let Some(request) = calculate_extension_request(
-                    usage.total_meta.sectors(),
-                    usage.used_meta.sectors(),
-                    meta_lowater.sectors(),
-                    false,
-                ) {
-                    match self.extend_thin_meta_device(pool_uuid, backstore, request) {
+                // Ensure meta blocks is 1/1000th of total usable size
+                let target_meta_size = backstore.datatier_usable_size() / 1000u16;
+                if usage.total_meta.sectors() < target_meta_size {
+                    let meta_request = target_meta_size - usage.total_meta.sectors();
```

Comment:
should we bump the meta size by some minimum?  maybe a MB?  otherwise we could trigger small meta resize.

Reply:
should be resolved by e5a8698